### PR TITLE
feat: add light client support to ckb full node

### PR DESCRIPTION
**Pull request by quake**
### What problem does this PR solve?

add light client support to ckb full node

### What is changed and how it works?

related rfcs:
[RFC(0043): CKB softfork activation](https://github.com/nervosnetwork/rfcs/pull/349)
[RFC: CKB Light Client Protocol](https://github.com/nervosnetwork/rfcs/pull/370)
[RFC: Client Side Block Filter Protocol](https://github.com/nervosnetwork/rfcs/pull/375)

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
Note: Add a note under the PR title in the release note.
```

### CODE REVIEW

File: verification/src/block_verifier.rs [hunk 1]
```diff
@@ -238,48 +237,6 @@ impl BlockBytesVerifier {
     }
 }
 
-/// BlockExtensionVerifier.
```
Comment (zhangsoledad) [inline verification/src/block_verifier.rs:241]: move to verification/contextual/src/contextual_block_verifier.rs

**Comment by zhangsoledad** [comment]
There are about 1w lines of new code that are generated, which can be excluded during the review, so don't be afraid.
截屏2022-10-11 15 03 44

File: spec/src/versionbits/mod.rs [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+use std::sync::Arc;
+
+/// What bits to set in version for versionbits blocks
+pub const VERSIONBITS_TOP_BITS: Version = 0x20000000;
+/// What bitmask determines whether versionbits is in use
+pub const VERSIONBITS_TOP_MASK: Version = 0xE0000000;
+/// Total bits available for versionbits
+pub const VERSIONBITS_NUM_BITS: u32 = 29;
+
+/// RFC0000 defines a finite-state-machine to deploy a soft fork in multiple stages.
```
Comment (zhangsoledad) [inline spec/src/versionbits/mod.rs:21]: A separate PR will follow to update the relevant documentation and parameters

**Review by zhangsoledad — APPROVED** [review]

**Review by yangby-cryptape — APPROVED** [review]

**Pull request merged**